### PR TITLE
Unit tests for MetricsActions component

### PR DESCRIPTION
**Pull request by Msarawan**

### CODE REVIEW

File: src/metrics/components/MetricsActions/MetricsActions.test.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { render, waitForI18n } from '../../../test-utils';
```
Comment (lavocatt) [inline src/metrics/components/MetricsActions/MetricsActions.test.tsx:1]: Why not importing directly from the react testing library?

File: src/metrics/components/MetricsActions/MetricsActions.tsx [hunk 2]
```diff
@@ -192,9 +192,9 @@ export const MetricsActions: FC<MetricsActionsProps> = ({
   ];
 
   return (
-    <Card isFullHeight>
+    <Card isFullHeight data-testid="metrics-card">
```
Comment (lavocatt) [inline src/metrics/components/MetricsActions/MetricsActions.tsx:195]: I'm not sure that we need to have `data-testid="metrics-card"` to find out the element when it is rendered.
You can use the playground to get a selector for that.

For instance, adding `screen.logTestingPlaygroundURL();` at line 21 will generate  [a screen url in the logs](https://testing-playground.com/#markup=DwEwlgbgfMCGBOAXMBjANgUwARhAXgCICt1YBnMwgBwDMBaFBhELWugWzpoFc006AFhjABzAYmIhYiWHUQYyyfAXYZE8VGSbwQk6bID23MLJQH2VAwDsMVxHICeVDIQAKAMQAsAegDCzPRk6IxM6MlgaFwJ1bgxAw2NTc0sbOzpcQgB5AFUASQBBOgBxWwx4aQwQOn8dOgBGAhhwCBI0ckoCNkYUZgB9XqFYEDLG0EhW9up6br7e2BRkazJ4uQUlQlV1TW0q+cWrZabxjJU1DRQtNDBFOhB4AyoQAwB3K2JSCimGW-vHl7esFIgiEkhZrLZ7IgnFEPD4ACK-J6vFYgsIRKIxOKA-TBRIMZLgtInHIFYqlcryKoIh5Iqz1UYAI24iEQ1mxQXkihOdxp-zkBhEIkwKLxZjBqUh0LcXm81L+rwAKgKhVigQlQuFIoRMSLQmKUhD0soSYUSjYKZU6HLaUrBZh6ThlJtzpdrvYefK7MrhRNPp1pj9ea9+qy7Vioc5CEyWdZiAhQhgAB5UWBWYbKGiwNBkLHx2QCciWKjcKja+CxUZkFNWX0dLqBz0h70YdJmN4wMgQERYRQOTCECBlZA9fhZ0RWABcWDoAAYAHR1ABMAFYMOwANzEGhgPiEFDceDwCG+AxoAzwYhCUTiQh1NfEZ64RACW-3rAQMAYZ4AIQMicIM5YIBADMi6AcuS5xho+a4MMVhlhWWD3P2BBgOwIijCmz6AoQACywF1HOwFYHUACci4CCuADsREoHUNEABxAVgi4AGxzlRLGEcuJGeAuWDAXxdQADL0UJAnLnxDGMIxWCMXQYFzjx8mLgxRFAaJNGLixbELr4fGsVxwFKSRxk8WR2kEepFkAF6jN4WECDA3idiIzlVqmznRqyVjOc0RwtCcHl0h6tLvG0fr1qF-y6qCBppBGMIytaMXsuqsiahi5aqjiqL6oS9jEnkprkhUVKInyi6MsyPlpasXLKNFrz8mGsX4uKhqJdK8IVYqzZtZlCE5cCooEhKRpZMVZLmmVVq9VYtoqgpxBBdWDa0qGKrhZM-rfE1VhNq1WBdQQ3mxlgeZ0Em1bpoQmbZrm0GCIWDwlkNlbVrWXyMPth1LfIiYSFAEWIL0wEzr07BgFYzIKMALnVh2n0fHWAa-b0m32qgsYdl2PZQihg5IKgWZ0GOIiTtO85LquG5bjuaB7geR52CeZ4XlgV5iIgr7sA+T4vgQd58++n4-n+AHMaB4GQRdT0CLBtjvUhp5RGhGEwI5OEEPhhHERZlHLjRwF0bJgG6Zxi7cbx-GCQumnicBklztJdCySp87KS7ClqcRM6aXO2mLrpdT6XOhlW2ZpkmRZAl6yR5F2c5jnuV27mI-DZ2+fD-ljIFyiWHw0MiOtqUo99pfInVeVjZ1UoELCsrzQN6LK2quJ6rXRLGlNZplLNKXNcB1UxjW7ecusBD7S1W3V6NHUJfXjeDwt-VzxqrfRNlbX5eNRWkn3FrlUGq9hnQw+OtQp5XFYJf7ZjWLl7tP3zX9PonVnUEJsmqa3QQ905i-vmF6xZSxbwrEjVMX1n6VwOhjZsqxAaNGXJDaGsMyDw2CpAmsT8oqv3gWfbG7ZgCuXxn2KIRNhyk3JpTWcC4Vxrk3Fgbcu4CD7kPMeU855LzCG5rzfmIBnz8NFl+X8-4CCARAmBLAEEqpy1CArEAcFlbITVuhTC0gBDa11tZcihtjamxdsxC2XEY6OyEg7W2ztXbu29opL2DEfbqX9vRQOOl2KhwMkZGOUdY5WX1oneyKdMFp0wRnbwWc-KQCidAHO0T4YIGHJgGJUAgA)



Then add some clicks from the user:
```
   const periodButton = screen.getByText(/last_30_minutes/i);
    expect(periodButton).toBeInTheDocument();
    userEvent.click(periodButton);
```

[You can see how it evolves](https://testing-playground.com/#markup=DwEwlgbgfMCGBOAXMBjANgUwARhAXgCICt1YBnMwgBwDMBaFBhELWugWzpoFc006AFhjABzAYmIhYiWHUQYyyfAXYZE8VGSbwQk6bID23MLJQH2VAwDsMVxHICeVDIQAKAMQAsAegDCzPRk6IxM6MlgaFwJ1bgxAw2NTc0sbOzpcQgB5AFUASQBBOgBxWwx4aQwQOn8dOgBGAhhwCBI0ckoCNkYUZgB9XqFYEDLG0EhW9up6br7e2BRkazJ4uQUlQlV1TW0q+cWrZabxjJU1DRQtNDBFOhB4AyoQAwB3K2JSCimGW-vHl7esFIgiEkhZrLZ7IgnFEPD4ACK-J6vFYgsIRKIxOKA-TBRIMZLgtInHIFYqlcryKoIh5Iqz1UYAI24iEQ1mxQXkihOdxp-zkBhEIkwKLxZjBqUh0LcXm81L+rwAKgKhVigQlQuFIoRMSLQmKUhD0soSYUSjYKZU6HLaUrBZh6ThlJtzpdrvYefK7MrhRNPp1pj9ea9+qy7Vioc5CEyWdZiAhQhgAB5UWBWYbKGiwNBkLHx2QCciWKjcKja+CxUZkFNWX0dLqBz0h70YdJmN4wMgQERYRQOTCECBlZA9fhZ0RWABcWDoAAYAHR1ABMAFYMOwANzEGhgPiEFDceDwCG+AxoAzwYhCUTiQh1NfEZ64RACW-3rAQMAYZ4AIQMicIM5YIBADMi6AcuS5xho+a4MMVhlhWWD3P2BBgOwIijCmz6AoQACywF1HOwFYHUACci4CCuADsREoHUNEABxAVgi4AGxzlRLGEcuJGeAuWDAXxdQADL0UJAnLnxDGMIxWCMXQYFzjx8mLgxRFAaJNGLixbELr4fGsVxwFKSRxk8WR2kEepFkAF6jN4WECDA3idiIzlVqmznRqyVjOc0RwtCcHl0h6tLvG0fr1qF-ysPQnBJtW6a6qCBppBGMIyta-zJWiWrROWqo4qi+qEvYxJ5Ka5IVFSiJ8oujLMj57KyJy6wENFrz8mGOUlRKjiRgQsKyrVirNjlmoYgVPUEn15WkmaZTVVaI1WLaKoKcQQXVg2tKhiq4WTP63wdVYTbdVg6VRo1sZYHmdAJam6YIbm0GCIWDwls9lbVrWXyMCdZ3rfIiYSFAEWIL0wEzr07BgFYzIKMALnVh2P0fHWAYA70e32qgsYdl2PZQihg5IKgWZ0GOIiTtO85LquG5bjuaB7geR52CeZ4XlgV5iIgr7sA+T4vgQd6C++n4-n+AHMaB4GQbdr0CLBthfUhp5RGhGEwI5OEEPhhHERZlHLjRwF0bJgG6Zxi7cbx-GCQumnicBklztJdCySp87Ke7ClqcRM6aXO2mLrpdT6XOhm22ZpkmRZAmGyR5F2c5jnuV27ko0j3nWDAvCK6EbQMhgfCVAyDiEMFO3-DjWLo39NfBjDtjcMQyFRKo8OjFc6soVY4KjLAF2wAycPDP+BAOmquJ6jNhqXYNmUrbk8jizPqITWrG+ivPRLGhVZLmktWWvKva4OhAWaxIQy7i3d4DhAymAZlmOYHZFmMrf0XfcOka-tw1hsVuYAAFg3IBDZcMM4YIzIEjWAzkrgwF7h3QgA8bBDxHmPNMSZCDTyKrvcUC8pRL3hCvABzVZ6yC3vlRCO855EP3lkQ+C0LQ1SDFYc+nB6rvmvlEOod8oKhEfqPF+hBMzZnrhFDGx1v4t3hv-N8qDTjw1AfecBihegCOgfDTk8DEFgGQWAPuUR0FxBgMPGQ2CJ54IaJQ4qe9JQDSGqfThFD6HUPRNvAhDDUplQPvNKqlJlocK4XQYCxAr5oBvgQKG99XoiOfpUcRb8pGHSinI3+ijxbKN-mowWGiIZQx0bA-RSMkHABQUAggZjMFWPHrgqediPH4kYU4jKZDQnuJ8Z4vKOp7GEL8UaZhgTj7BNcWEzwkS+G3gEEI2QiSxEEAke-X6R1-qZNbtkwBKE8lgPBlogYRh4BlO8BUqp-dB4WKwQ0ye+DgSDNKv1Dpw0ulvhaTQ-pLTeqGjmpVMZloJlrzoMuaZ0SoisTmYXBZ1xRHJOWakj+MiNkcJ-ls-JOzO4gP2RA3orEjkHjgd4BB5TDGVOMco2p1z6k4Luc0nprShmLxceQ95DLPlTQGb4p5fyj6LXGayzgrEwUxKXFCh+sKkmv0kUixuWMskYpMcA1ROLNFLgJfAIlJKzlkouaYq5cAbm0tsdNNpzzpSdM9Fw8aXjaGFQedy2aAT-n8sBYKz2Ir+G6GhbcSVSyVlpM-rI1F8i-6Ktydi9RBy6i9CkA4U55yKXVKpYamlNimmmqZSQllbz17sttV8hlPymEEBNHythISrXAoYp6wgi5vUSqfv6xFayMkhoVRQiNKqo24sXLG2ADgtUGKMUqmpBrLGj1uSarlKUnnMuXrmm1fTOXfMccM0tLCglutzXQUitbRbPHmb6pt8KA2yvWU3U6obtmjr2T2tVvRngYAwAAawTbqpNlyMHUsncajNM7GVzuzQuqtbKHW9MmnQota7eWsJPu6uoM592LkPT6xZp6W0NwvfK9Fnbql3oKQcvtT7X1DtJc5XgflIABUdNQU8VwrAiEveettnpM1AecSBsKAGOVQfA4Bp1IyXUVtceEhqMYawz1atyFaXV9oAeLe0i1rzPRrR9B8gtK7oNmtg1u9hqnmxido50ejcMmMnTrixr+IbLMXRIbnAEd0Ho4Olasu6BYqwfVLHa76qZW3WcbNjQzwNQZQNhroxGyNPLAGCv54NgW66tnxjFwmvYSZDnJqOK41MpyzgXCuNcm4sDbl3AQfch5jynnPJeYQfMBZCxAM+erEsvy-knoBECYEsAQR4e5lW8EfOjq1phaQAg9YG2suRE2ZsLbu2YtbLi8cXZCWdg7N2HsvZ+0Ur7Bi-t1JB3oiHHS7EI4GSMvHWOCcrJGxTvZdOSNXJZ2i94BzVHoBI38h96j8CyboAwG9qAQA)

**Review by lavocatt — CHANGES_REQUESTED** [review]
This is a good start but, I'd like the test to do more than just attest the existence of the component. Like clicking on the various buttons available and check that the actual functionality at play.

For instance on this component, when the user changes the period, the component should update accordingly. Let's check that out.

File: src/metrics/components/MetricsActions/MetricsActions.test.tsx (hunk 1 above)
Comment (Msarawan) [reply src/metrics/components/MetricsActions/MetricsActions.test.tsx:1]: @lavocatt , that also we can do. but during the test setup we created the test-utils file and added all the test related export in that file and whenever we need any import we just import the test-utils. files instead separately importing.
i.e (export * from '@testing-library/react';)

File: src/metrics/components/MetricsActions/MetricsActions.tsx (hunk 2 above)
Comment (lavocatt) [reply src/metrics/components/MetricsActions/MetricsActions.tsx:195]: At first my opinion was that adding `data-testid ` would clutter the tags. But I think that after a bit of research, it looks like the proper way of doing it.

Comment (lavocatt) [reply src/metrics/components/MetricsActions/MetricsActions.tsx:195]: But interestingly there's an article on testing library's website we use that is saying that we should consider `data-test-id` in last resort  https://testing-library.com/docs/queries/about/#priority

Comment (lavocatt) [reply src/metrics/components/MetricsActions/MetricsActions.tsx:195]: So I'd say, let's do a bit of learning on the testing library and then choose the recommended option.

**Review by lavocatt — APPROVED** [review]

**Review by gaohoward — APPROVED** [review]

**Pull request merged**